cmd/eksctl-anywhere/cmd/installpackagecontroller.go

```diff
@@ -37,31 +40,43 @@ var installPackageControllerCommand = &cobra.Command{
 }
 
 func runInstallPackageController(cmd *cobra.Command, args []string) error {
-	if err := curatedpackages.ValidateKubeVersion(ico.kubeVersion, curatedpackages.Registry); err != nil {
-		return err
+	clusterConfigFileExist := validations.FileExists(ico.fileName)
+	if !clusterConfigFileExist {
+		return fmt.Errorf("the cluster config file %s does not exist", ico.fileName)
 	}
-
 	return installPackageController(cmd.Context())
 }
 
 func installPackageController(ctx context.Context) error {
 	kubeConfig := kubeconfig.FromEnvironment()
 
-	deps, err := curatedpackages.NewDependenciesForPackages(ctx, kubeConfig)
+	clusterSpec, err := v1alpha1.GetAndValidateClusterConfig(ico.fileName)
+	if err != nil {
+		return fmt.Errorf("the cluster config file provided is invalid: %v", err)
+	}
+
+	deps, err := curatedpackages.NewDependenciesForPackages(ctx)
 	if err != nil {
 		return fmt.Errorf("unable to initialize executables: %v", err)
 	}
 
-	versionBundle, err := curatedpackages.GetVersionBundle(deps.ManifestReader, version.Get().GitVersion, ico.kubeVersion)
+	versionBundle, err := curatedpackages.GetVersionBundle(deps.ManifestReader, version.Get().GitVersion, clusterSpec)
 	if err != nil {
 		return err
 	}
+
+	registryEndpoint := ""
+	if clusterSpec.Spec.RegistryMirrorConfiguration != nil {
+		registryEndpoint = clusterSpec.Spec.RegistryMirrorConfiguration.Endpoint
+	}
+
 	helmChart := versionBundle.PackageController.HelmChart
+	imageUrl := urls.ReplaceHost(helmChart.Image(), registryEndpoint)
```

Comment:
Does this work if the registryEndpoint is empty string?

Reply:
Yes, ReplaceHost returns the same url if empty string.